### ISSUE TITLE
[Triton] Timm_models  rexnet_100 / fbnetv3_b / sebotnet33ts_256 got fail_accuracy

### ISSUE THREAD

**Issue by kaileiyx** [intel/torch-xpu-ops#3151]
### 🐛 Describe the bug

With new triton 3.7.1+git21033c4e got fail_accuracy. But with 64bb0de3 got pass.
Reproducer:
```
docker run -it --device=/dev/mem --device=/dev/dri --group-add video --privileged --shm-size=8g intelgpu/ubuntu-24.04-rolling:26.05 bash

uv venv myvenv --python 3.10 --clear
source myvenv/bin/activate
uv pip install pip 'setuptools<81' wheel

uv pip install torch torchvision torchaudio torchao --index-url https://download.pytorch.org/whl/test/xpu
uv pip install pip pandas psutil scipy requests
uv pip install -U numpy==1.26.4
uv pip install transformers==4.55.2 accelerate

pytorch_commit="$(python -c 'import torch; print(torch.version.git_version)')"
git clone https://github.com/pytorch/pytorch
cd pytorch
git checkout ${pytorch_commit}

rm -rf ~/.cache/ ~/.triton/ /tmp/torchinductor_root/
python benchmarks/dynamo/timm_models.py --accuracy -d xpu -n10 --backend=inductor --cold-start-latency --training --amp --amp-dtype float16 --only fbnetv3_b
python benchmarks/dynamo/timm_models.py --accuracy -d xpu -n10 --backend=inductor --cold-start-latency --training --amp --amp-dtype float16 --only sebotnet33ts_256
python benchmarks/dynamo/timm_models.py --accuracy -d xpu -n10 --backend=inductor --cold-start-latency --training --amp --amp-dtype bfloat16 --only rexnet_100

```
errorlog:
```
(triton) sdp@emr7470:~/kaileiyx/pytorch$ python benchmarks/dynamo/timm_models.py --accuracy -d xpu -n10 --backend=inductor --cold-start-latency --training --amp --amp-dtype float16 --only fbnetv3_b
loading model: 0it [00:05, ?it/s]
xpu  train fbnetv3_b
E0324 03:04:55.713000 1710171 site-packages/torch/_dynamo/utils.py:3393] RMSE (res-fp64): 0.00724, (ref-fp64): 0.00073 and shape=torch.Size([16]). res.dtype: torch.float32, multiplier: 3.000000, tol: 0.010000, use_larger_multiplier_for_smaller_tensor: 0
E0324 03:04:55.714000 1710171 site-packages/torch/_dynamo/utils.py:3231] Accuracy failed for key name blocks.0.0.bn1.weight.grad
fail_accuracy

(triton) sdp@emr7470:~/kaileiyx/pytorch$ python benchmarks/dynamo/timm_models.py --accuracy -d xpu -n10 --backend=inductor --cold-start-latency --training --amp --amp-dtype bfloat16 --only rexnet_100
loading model: 0it [00:08, ?it/s]
xpu  train rexnet_100
E0324 03:10:11.243000 1714504 site-packages/torch/_dynamo/utils.py:3393] RMSE (res-fp64): 0.00181, (ref-fp64): 0.00031 and shape=torch.Size([1044]). res.dtype: torch.float32, multiplier: 2.000000, tol: 0.010000, use_larger_multiplier_for_smaller_tensor: 0
E0324 03:10:11.243000 1714504 site-packages/torch/_dynamo/utils.py:3231] Accuracy failed for key name features.15.conv_dw.bn.running_var
fail_accuracy


(triton) sdp@emr7470:~/kaileiyx/pytorch$ python benchmarks/dynamo/timm_models.py --accuracy -d xpu -n10 --backend=inductor --cold-start-latency --training --amp --amp-dtype float16 --only sebotnet33ts_256
loading model: 0it [00:05, ?it/s]
xpu  train sebotnet33ts_256
E0324 03:14:55.508000 1722167 site-packages/torch/_dynamo/utils.py:3393] RMSE (res-fp64): 2.23511, (ref-fp64): 0.95341 and shape=torch.Size([128, 128, 3, 3]). res.dtype: torch.float32, multiplier: 2.000000, tol: 0.010000, use_larger_multiplier_for_smaller_tensor: 0
E0324 03:14:55.508000 1722167 site-packages/torch/_dynamo/utils.py:3231] Accuracy failed for key name stages.1.1.conv2_kxk.conv.weight.grad
fail_accuracy

```

### Versions

torch                  2.11.0+xpu
torchao                0.16.0+xpu
torchaudio             2.11.0+xpu
torchvision            0.26.0+xpu
triton-xpu             3.7.1+git21033c4e

**Comment by chuanqi129** [comment]
Triton team decided to workaround the issue temporarily by reverting parts of ReduceOpToLLVM that cause accuracy differences (https://github.com/intel/intel-xpu-backend-for-triton/pull/6497), opened https://github.com/intel/intel-xpu-backend-for-triton/issues/6719 for further investigation.

**Comment by chuanqi129** [comment]
Waiting for new PT 2.12 RC to do the verification

**Comment by kaileiyx** [comment]
still failed for pt2.12 rc test.  [#3465](https://github.com/intel/torch-xpu-ops/issues/3465)

**Comment by chuanqi129** [comment]
Hi @kaileiyx let‘s use the latest RC and try it again

**Comment by kaileiyx** [comment]
@chuanqi129  Using the latest RC, all tests passed.
Platform: BMG B60
OS: Ubuntu 25.10
pytorch commit id: 54f2efc1073c20fe00e5ae57dfe9123c658c3ea0
```
(ww17) sdp@emr7470:~/kaileiyx/pytorch$ python benchmarks/dynamo/timm_models.py --accuracy -d xpu -n10 --backend=inductor --cold-start-latency --training --amp --amp-dtype bfloat16 --only rexnet_100
model.safetensors: 100%|█████████████████████████████████████████████████████████████| 19.4M/19.4M [00:01<00:00, 16.0MB/s]
loading model: 0it [00:08, ?it/s]████████████████████████████████████████████████████| 19.4M/19.4M [00:01<00:00, 96.5MB/s]
xpu  train rexnet_100
pass
(ww17) sdp@emr7470:~/kaileiyx/pytorch$ python benchmarks/dynamo/timm_models.py --accuracy -d xpu -n10 --backend=inductor --cold-start-latency --training --amp --amp-dtype float16 --only fbnetv3_b
model.safetensors: 100%|█████████████████████████████████████████████████████████████| 34.6M/34.6M [00:01<00:00, 28.6MB/s]
loading model: 0it [00:06, ?it/s]█████████████████████████████████████████████████████| 34.6M/34.6M [00:01<00:00, 172MB/s]
xpu  train fbnetv3_b
pass
(ww17) sdp@emr7470:~/kaileiyx/pytorch$ python benchmarks/dynamo/timm_models.py --accuracy -d xpu -n10 --backend=inductor --cold-start-latency --training --amp --amp-dtype float16 --only sebotnet33ts_256
model.safetensors: 100%|█████████████████████████████████████████████████████████████| 55.0M/55.0M [00:01<00:00, 30.3MB/s]
loading model: 0it [00:07, ?it/s]█████████████████████████████████████████████████████| 55.0M/55.0M [00:01<00:00, 274MB/s]
xpu  train sebotnet33ts_256
pass
```
Version
```
torch              2.12.0+xpu
torchao            0.17.0+xpu
torchaudio         2.11.0+xpu
torchvision        0.27.0+xpu
triton-xpu         3.7.1
```
